catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/operation/impl/ProcessingDetailsImpl.java

```diff
@@ -82,6 +93,21 @@ public ProcessingDetailsImpl(SourceProcessingDetails details, String sourceId) {
     return warnings;
```

Comment:
✏️ Could use `Collections.singletonList(...)` instead.

Reply:
...instead of to `toWarningList()`?

Reply:
Correct. The `toWarningList()` method is creating a general purpose collection, meant to hold multiple elements, and is mutable. By using `Collections.singletonList(list)` you are cutting down on code, memory usage, and get immutability for free.